mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -1161,6 +1186,30 @@ void mav_vehicle::update()
             }
         }
     }
+
+    // Check if autorotation is active in order to request rotation if needed,
+    // if not already rotating. Need also to make sure that a mission waypoint
+    // is not currently being sent.
+    if (this->waypoint_autorotate &&
+        (this->mstatus == mission_status::NORMAL ||
+         this->mstatus == mission_status::DETOURING) &&
+        !this->sending_mission) {
```

Comment:
Indeed, we might need to add more code to this function later. I don't like those huge boolean expressions either, but I couldn't find a way of doing it neatly. Anyway, I'll take another look into it.